src/main/java/org/semver4j/internal/Modifier.java

```diff
@@ -11,111 +11,179 @@
 import static java.util.Collections.emptyList;
 
 public class Modifier {
-    @NotNull
-    private final Semver version;
+    private static final String FULL_FORMAT = "%d.%d.%d";
+    private static final String MAJOR_FORMAT = "%d.0.0";
+    private static final String MAJOR_MINOR_FORMAT = "%d.%d.0";
 
-    public Modifier(@NotNull final Semver version) {
-        this.version = version;
+    private Modifier() {
     }
 
     @NotNull
-    public Semver nextMajor() {
+    public static Semver nextMajor(@NotNull final Semver version) {
         int nextMajor = version.getMajor();
 
         // Prerelease version 1.0.0-5 bumps to 1.0.0
         if (version.getMinor() != 0 || version.getPatch() != 0 || version.getPreRelease().isEmpty()) {
             nextMajor = nextMajor + 1;
         }
 
-        String version = createFullVersion(format(Locale.ROOT, "%d.0.0", nextMajor), emptyList());
-        return new Semver(version);
+        return new Semver(createFullVersion(version, format(Locale.ROOT, MAJOR_FORMAT, nextMajor), emptyList()));
     }
 
     @NotNull
-    public Semver withIncMajor(int number) {
-        String version = createFullVersion(format(Locale.ROOT, "%d.%d.%d", (this.version.getMajor() + number), this.version.getMinor(), this.version.getPatch()), this.version.getPreRelease());
-        return new Semver(version);
+    public static Semver withIncMajor(@NotNull final Semver version, int number) {
+        return new Semver(
+                createFullVersion(
```

Comment:
Why you resolved this and the following comments?

Reply:
Because I reverted the change that provoked the comments?

Reply:
If the comments referred to another point, please let me know, because I interpreted "keep formatting as it's introduced" as "do not memoize the formatting string used in each invocation of `createFullVersion()`".

Reply:
Misunderstood, I would like to maintain the original format of the variable `version`, which is extracted onto a separate line.